packages/server/lib/controllers/connect/postSessions.ts

```diff
@@ -0,0 +1,121 @@
+                    .record(
+                        z.object({
+                            connectionConfig: z.record(z.unknown())
+                        })
+                    )
+                    .optional()
+            })
+            .strict()
+            .parse(data)
+});
+
+const handler = asyncWrapper<PostConnectSessions>(async (req, res) => {
+    await db.knex.transaction(async (trx) => {
+        // Check if the linkedProfile exists in the database
+        const getLinkedProfile = await linkedProfileService.getLinkedProfile(trx, {
+            profileId: req.body.linkedProfile.profileId,
+            accountId: res.locals.account.id,
+            environmentId: res.locals.environment.id
+        });
+
+        let linkedProfileId: number;
+        if (getLinkedProfile.isErr()) {
+            // create linked profile if it doesn't exist yet
+            if (getLinkedProfile.error.code === 'not_found') {
+                // fail if linkedProfile doesn't exist and email is not provided
+                if (!req.body.linkedProfile.email) {
+                    res.status(400).send({
+                        error: {
+                            code: 'invalid_request',
+                            errors: [{ code: 'invalid_type', message: 'email is required', path: ['linkedProfile', 'email'] }]
+                        }
+                    });
+                    return;
+                }
+
+                const createLinkedProfile = await linkedProfileService.createLinkedProfile(trx, {
+                    profileId: req.body.linkedProfile.profileId,
+                    email: req.body.linkedProfile.email,
+                    displayName: req.body.linkedProfile.displayName || null,
+                    organization: req.body.linkedProfile.organization?.organizationId
+                        ? {
+                              organizationId: req.body.linkedProfile.organization.organizationId,
+                              displayName: req.body.linkedProfile.organization.displayName || null
+                          }
+                        : null,
+                    accountId: res.locals.account.id,
+                    environmentId: res.locals.environment.id
+                });
+                if (createLinkedProfile.isErr()) {
+                    res.status(500).send({ error: { code: 'internal_error', message: 'Failed to create linked profile' } });
+                    return;
+                }
+                linkedProfileId = createLinkedProfile.value.id;
+            } else {
+                console.log(getLinkedProfile.error);
+                res.status(500).send({ error: { code: 'internal_error', message: 'Failed to get linked profile' } });
+                return;
+            }
+        } else {
+            // TODO: what if linkedProfile exists but email, displayName, or organization is different?
+            linkedProfileId = getLinkedProfile.value.id;
```

Comment:
as we said in the meeting, profileId should probably be `companyId + profileId` 
if the email change then it needs update, if the companyId is different that means it's a different profile (even if it's the same end user)

Reply:
let's not call it profileId if it is not an id then. It might be confusing to users if we consider `profileId + orgId` to be the unique key